src/probability/probability_mass_function/basic.lean

```diff
@@ -55,6 +58,13 @@ def support (p : pmf α) : set α := function.support p
 
 @[simp] lemma mem_support_iff (p : pmf α) (a : α) : a ∈ p.support ↔ p a ≠ 0 := iff.rfl
 
+lemma support_nonempty (p : pmf α) : p.support.nonempty :=
+begin
+  refine set.nonempty_def.2 (by_contra $ λ h, _),
+  simp only [pmf.mem_support_iff, not_exists, not_not] at h,
+  exact zero_ne_one (((tsum_congr h).trans tsum_zero).symm.trans p.tsum_coe),
+end
```

Comment:
```suggestion
function.support_nonempty_iff.2 a_missing_lemma
```

Reply:
I named this additional lemma as `coe_fn_ne_zero` which seemed natural to me, however the naming convention for `tsum_coe` and some functions above doesn't include the `fn` portion. Not sure if these should match for consistency.